userspace/libgolf2/aes_ecb_syscalls.h

```diff
@@ -0,0 +1,92 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "tock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Configures the encryption key to be used for encryption and decryption.
+//
+// key - a buffer containing the key (must be 16 bytes)
+// len - length of the buffer (must be 16 bytes)
+int tock_aes128_set_key(const unsigned char* key, unsigned char len);
+
+
+// Encrypts a payload according to aes-128 counter-mode. The caller is
+// responsible for incrementing/managing the counter.
+//
+// buf      - buffer to encrypt (should be <= 16 bytes, will be zero-padded)
+// buf_len  - length of the buffer to encrypt
+// ctr      - buffer with the initial counter (should be 16 bytes)
+// ctr_len  - length of buffer with the initial counter (should be 16 bytes)
+int tock_aes128_encrypt_ctr_sync(unsigned char* buf, unsigned char buf_len,
+                                 unsigned char* ctr, unsigned char ctr_len);
+
+// Decrypts a payload according to aes-128 counter-mode. The caller
+// is responsible for incrementing/managing the counter.
+//
+// buf      - buffer to decrypt (should be <= 16 bytes, will be zero-padded)
+// buf_len  - length of the buffer to decrypt
+// ctr      - buffer with the initial counter (should be 16 bytes)
+// ctr_len  - length of buffer with the initial counter (should be 16 bytes)
+int tock_aes128_decrypt_ctr_sync(unsigned char* buf, unsigned char buf_len,
+                                 unsigned char* ctr, unsigned char ctr_len);
+
+// Encrypts a payload according to aes-128 electronic codebook mode. Note
+// that this encryption mode is generally frowned upon, two identical
+// cleartexts have the same ciphertext (it leaks information).
+//
+// buf      - buffer to encrypt (should be <= 16 bytes, will be zero-padded)
+// buf_len  - length of the buffer to encrypt
+// ctr      - buffer with the initial counter (should be 16 bytes)
+// ctr_len  - length of buffer with the initial counter (should be 16 bytes)
+int tock_aes128_encrypt_ecb_sync(unsigned char* buf, unsigned char buf_len);
+
+// Decrypts a payload according to aes-128 electronic codebook mode. Note
+// that this encryption mode is generally frowned upon, two identical
+// cleartexts have the same ciphertext (it leaks information).
+//
+// buf      - buffer to decrypt (should be <= 16 bytes, will be zero-padded)
+// buf_len  - length of the buffer to decrypt
+// ctr      - buffer with the initial counter (should be 16 bytes)
+// ctr_len  - length of buffer with the initial counter (should be 16 bytes)
+int tock_aes128_decrypt_ecb_sync(unsigned char* buf, unsigned char buf_len);
+
+
+// Encrypts a payload according to aes-128 cipher-block-chaining mode.
+//
+// buf      - buffer to encrypt (should be <= 16 bytes, will be zero-padded)
+// buf_len  - length of the buffer to encrypt
+// iv       - the initialization vector (must be 16 bytes)
```

Comment:
Presumably the caller is responsible for forwarding the previous ciphertext through to the next block's IV? If so, that should be documented.

Reply:
Yes. Will do.